docs/servo_power_monitor/src/power_test_controller.ts

```diff
@@ -75,26 +73,29 @@ export class PowerTestController {
           this.powerDataList[this.powerDataList.length - 1][0],
           'end'
         );
+      } else if (dutData.includes('stop')) {
+        await this.stopMeasurement();
       }
-      this.ui.addMessageToConsole(dutData);
+      await this.ui.addMessageToConsole(dutData);
     }
   }
   public async startMeasurement() {
     await this.runner.setScript();
-    await this.servoController.servoShell.open();
-    this.changeHaltFlag(false);
+    await this.servoController.openServoPort();
+    await this.runner.openDutPort();
+    await this.changeHaltFlag(false);
+    this.readDutLoop();
     this.kickWriteLoop();
     this.readLoop();
+    await this.runner.copyScriptToDut();
+    await this.runner.executeScript();
   }
-  public async stopMeasurement() {
+  public async stopMeasurement(abortFlag = false) {
+    if (abortFlag) await this.runner.sendCancel();
```

Comment:
Does the function need an abort flag? How about just calling sendCancel() when this.inProgress is true at the beginning of this function?

Reply:
I added this `sendCancel()` to finish the running command on DUT console, but `this.inProgress` and `this.halt` flag control the read/write loop on servo. So I think whether `this.inProgress` is true or false doesn't matter. However, I agree that it doesn't `abortFlag` and we can just call `sendCancel()` regardless of whether some command is running or not on DUT.

Reply:
I'm not sure whether `await this.runner.sendCancel()` finishes even when the case which was abortFlag == false. Did you check that case...? (I'm just curious and it may be a rare error case so it may not be a huge matter but I'm curious if you checked that. thanks!!)

Reply:
Discussed with @wtrgjadmw offline, and we decided to go with the current implementation (call sendCancel anyway even if it is already closed) since it seems to be okay for now and simplify the implementation.